lib/vrl/web-playground/public/index.html

```diff
@@ -88,83 +88,150 @@ <h1>[draft] VRL playground wasm</h1>
-      });
-
-      window.eventEditor = monaco.editor.create(document.getElementById('container-event'), {
-        value: `{
+          language: "coffeescript",
+          theme: "vs-light",
+          // minimap: { enabled: false },
+          automaticLayout: true,
+        });
+
+        window.eventEditor = monaco.editor.create(document.getElementById("container-event"), {
+          value: `{
   "message": "<102>1 2020-12-22T15:22:31.111Z vector-user.biz su 2666 ID389 - Something went wrong"
 }`,
-        language: 'json',
-        theme: 'vs-light',
-        minimap: { enabled: false }
+          language: "json",
+          theme: "vs-light",
+          // minimap: { enabled: false },
+          automaticLayout: true,
+        });
+
+        window.outputEditor = monaco.editor.create(document.getElementById("container-output"), {
+          language: "json",
+          theme: "vs-light",
+          // minimap: { enabled: false },
+          automaticLayout: true,
+        });
+        const queryString = window.location.search;
+        if (queryString.length != 0) {
+          const urlParams = new URLSearchParams(queryString);
+          const stateParam = decodeURIComponent(urlParams.get("state"));
+
+          try {
+            let urlState = JSON.parse(atob(stateParam));
+
+            window.programEditor.setValue(urlState["program"]);
+            window.eventEditor.setValue(JSON.stringify(urlState["event"], null, "\t"));
+
+            console.log("[DEBUG::queryStringLogic] Current Params:", JSON.parse(atob(stateParam)));
+            let res = handleRunCode(JSON.parse(atob(stateParam)));
+            console.log("[DEBUG::queryStringLogic] Running VRL with current Params:", res);
+          } catch (e) {
+            window.outputEditor.setValue(`Error reading the shared URL\n${e}`);
+          }
+        }
       });
+    </script>
 
-      window.outputEditor = monaco.editor.create(document.getElementById('container-output'), {
-        language: 'json',
-        theme: 'vs-light',
-        minimap: { enabled: false }
+    <script type="module">
+      import init, { run_vrl } from "./vrl_web_playground.js";
+      init().then(() => {
+        window.run_vrl = run_vrl;
       });
-      const queryString = window.location.search;
-      if (queryString.length != 0) {
-        const urlParams = new URLSearchParams(queryString);
-        const stateParam = urlParams.get('state');
-
-        let urlState = JSON.parse(atob(stateParam));
-
-        window.programEditor.setValue(urlState["program"]);
-        window.eventEditor.setValue(JSON.stringify(urlState["event"], null, '\t'));
-
-        console.log("[DEBUG::queryStringLogic] Current Params:", JSON.parse(atob(stateParam)));
-        let res = handleRunCode(JSON.parse(atob(stateParam)));
-        console.log("[DEBUG::queryStringLogic] Running VRL with current Params:", res);
+    </script>
+
+    <script>
+      function tryJsonParse(str) {
```

Comment:
we add a general jsonparse to display an error in case the editor contents cannot be parsed by json, this prevents users not seeing any feedback and having to open up the console to see what's wrong. 

we share common errors mistakes here as well